src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,207 @@
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturers_id = ? "
+                + "WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateQuery)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update info from db", e);
+        }
+        deleteDriver(car);
+        addDriver(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carID) {
+        String updateQuery = "UPDATE cars SET is_deleted = TRUE AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateQuery)) {
+            statement.setLong(1, carID);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant update info from db", e);
+        }
+    }
+
+    @Override
+    public void addDriver(Car car) {
+        String addDriverRequest = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(addDriverRequest)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver: car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add driver from db", e);
+        }
+    }
+
+    @Override
+    public void deleteDriver(Car car) {
+        String removeRequest = "DELETE FROM cars_drivers  "
+                + "WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(removeRequest)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't remove driver from db", e);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarRequest = "SELECT c.id, model, manufacturers_id, name, country  "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd ON c.id = car_id "
+                + "JOIN manufacturers m ON c.manufacturers_id = m.id "
+                + "WHERE cd.driver_id = 3 AND c.is_deleted = FALSE;";

```

Comment:
why driver id = 3?

Reply:
my mistake !!! made an inspection in Mysql